lib/avromatic/model/types/union_type.rb

```diff
@@ -12,6 +12,10 @@ def initialize(member_types:)
           @value_classes = member_types.flat_map(&:value_classes)
         end
 
+        def name
+          "union[#{member_types.map(&:name).join(', ')}]"
```

Comment:
Is it worth memoizing the non-constant name methods?

Reply:
I didn't bother since this method isn't called for a hot codepath. We can always add it later if it becomes a bottleneck.